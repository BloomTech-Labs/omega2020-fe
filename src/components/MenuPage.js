import React from 'react';
import { NavLink } from 'react-router-dom';

function MenuPage() {
  return (
    <div>
      <h1>Omega2020 Menu Page</h1>
    </div>
  );
}

export default MenuPage;