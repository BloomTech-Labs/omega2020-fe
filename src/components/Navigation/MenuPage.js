import React from 'react';
import Box from '@material-ui/core/Box';
import Link from '@material-ui/core/Link';
import Skeleton from '@material-ui/lab/Skeleton';
import CardHeader from '@material-ui/core/CardHeader';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const MenuPage = (props) => {
  const classes = useStyles();
  const { loading = false } = props;

  return (
    <Box className={classes.root}>
      <Box className={classes.mainContainer}>
        <CardHeader
          avatar={
            loading ? (
              <Skeleton
                className='bannerIMG'
                variant='rect'
                width={600}
                height={600}
              />
            ) : (
              <Box className={classes.headerBox}>
                <Typography
                  className={classes.header}
                  style={{ fontSize: '1.3rem' }}
                >
                  MENU
                </Typography>

                <br />
                <br />

                <Link href='/'>
                  <Typography className={classes.subheader}>Home</Typography>
                </Link>
                <br />
                <Link href='/levels'>
                  <Typography className={classes.subheader}>Play</Typography>
                </Link>
                <br />
                <Link href='/about'>
                  <Typography className={classes.subheader}>About</Typography>
                </Link>
                <br />
                <Link href='/tutorial'>
                  <Typography className={classes.subheader}>
                    Tutorials
                  </Typography>
                </Link>
                <br />
              </Box>
            )
          }
        />
      </Box>
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    // height: '80vh',
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 200,
  },
  mainContainer: {
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
  subheader: {
    // opacity: 0.3,
    color: '#A2A2A2',
  },
  space: {
    '& > *': {
      margin: theme.spacing(1),
    },
  },
}));

export default MenuPage;
