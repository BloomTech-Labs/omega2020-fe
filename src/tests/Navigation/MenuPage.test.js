import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom/extend-expect';
import { Box, Typography, Link, CardHeader } from '@material-ui/core';
import MenuPage from '../../components/Navigation/MenuPage';

configure({ adapter: new Adapter() });


describe('the length of the material ui and match the snapshot of the foot mobile component', () => {

    const wrapper = mount(<MenuPage />)

    it('snapshot the component', () => {
        const menuShot = renderer.create(<MenuPage />).toJSON();
        expect(menuShot).toMatchSnapshot();
    })

    it('render the length of the material ui', () => {
        expect(wrapper.find(CardHeader).length).toEqual(1);
        expect(wrapper.find(Typography).length).toEqual(7);
        expect(wrapper.find(Box).length).toEqual(3);
        expect(wrapper.find(Link).length).toEqual(3);
    })


    it('renders all the text contents in the component', () => {
        expect(wrapper.containsMatchingElement(
            <Typography>
                MENU
            </Typography>)
        ).toBeTruthy();
        
        expect(wrapper.containsMatchingElement(
            <Typography>
                Home
            </Typography>
        )).toBeTruthy();

        expect(wrapper.containsMatchingElement(
            <Typography>
                About
            </Typography>
        )).toBeTruthy();

        expect(wrapper.containsMatchingElement(
            <Typography>
                Tutorials
            </Typography>
        )).toBeTruthy();

    })
})