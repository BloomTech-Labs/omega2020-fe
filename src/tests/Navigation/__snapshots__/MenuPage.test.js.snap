// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the length of the material ui and match the snapshot of the foot mobile component snapshot the component 1`] = `
<div
  className="MuiBox-root MuiBox-root-8 makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-9 makeStyles-mainContainer-2"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          className="MuiBox-root MuiBox-root-10"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
            style={
              Object {
                "fontSize": "1.3rem",
              }
            }
          >
            MENU
          </p>
          <br />
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <p
              className="MuiTypography-root makeStyles-subheader-3 MuiTypography-body1"
            >
              Home
            </p>
          </a>
          <br />
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/about"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <p
              className="MuiTypography-root makeStyles-subheader-3 MuiTypography-body1"
            >
              About
            </p>
          </a>
          <br />
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/tutorial"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <p
              className="MuiTypography-root makeStyles-subheader-3 MuiTypography-body1"
            >
              Tutorials
            </p>
          </a>
          <br />
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      />
    </div>
  </div>
</div>
`;
